docs/examples/good_practices/wandb_setup/main.py

```diff
@@ -0,0 +1,194 @@
+"""Single-GPU training example."""
+import logging
+import os
+from pathlib import Path
+
+import rich.logging
+import torch
+from torch import Tensor, nn
+from torch.nn import functional as F
+from torch.utils.data import DataLoader, random_split
+from torchvision import transforms
+from torchvision.datasets import CIFAR10
+from torchvision.models import resnet18
+from tqdm import tqdm
+import wandb
+
+def main():
+    training_epochs = 10
+    learning_rate = 5e-4
+    weight_decay = 1e-4
+    batch_size = 128
+
+    # Check that the GPU is available
+    assert torch.cuda.is_available() and torch.cuda.device_count() > 0
+    device = torch.device("cuda", 0)
+
+    # Setup logging (optional, but much better than using print statements)
+    logging.basicConfig(
+        level=logging.INFO,
+        handlers=[rich.logging.RichHandler(markup=True)],  # Very pretty, uses the `rich` package.
+    )
+
+    logger = logging.getLogger(__name__)
+
+    # Setup Wandb
+    run = wandb.init(
+    # Set the project where this run will be logged
+    project="awesome-wandb-example",
+    name=os.environ.get("SLURM_JOB_ID"),
+    resume="allow", # See https://docs.wandb.ai/guides/runs/resuming
+    # Track hyperparameters and run metadata
+    config={
+        "learning_rate": learning_rate,
+        "epochs": training_epochs,
+        "weight_decay": weight_decay,
+        "batch_size": batch_size,
+    })
```

Comment:
Fixing the indentation:
```suggestion
    run = wandb.init(
        # Set the project where this run will be logged
        project="awesome-wandb-example",
        name=os.environ.get("SLURM_JOB_ID"),
        resume="allow", # See https://docs.wandb.ai/guides/runs/resuming
        # Track hyperparameters and run metadata
        config={
            "learning_rate": learning_rate,
            "epochs": training_epochs,
            "weight_decay": weight_decay,
            "batch_size": batch_size,
        },
    )
```